### PR TITLE
Fix CDATA encoding issue

### LINKED ISSUE #7260: RSS Feed double encodes CDATA
The `FeedController` double encodes CDATA when it doesn't need too. For example, viewing source nodes using CDATA should look like this:

`<description><![CDATA[<p>My amazing description.</p>]]></description>`

Instead it is double encoded like this:
`<description>&lt;![CDATA[&lt;p&gt;My amazing description.&lt;/p&gt;]]&gt;</description>`

This messes up some feed readers which results in them displaying the description like: "My amazing description.]]"

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #7260

- Avoid double encode CDATA
- Use XData

### CODE REVIEW

**Comment by hishamco** [comment]
I may add a unit test for this one

**Comment by Skrypt** [comment]
Looks good

File: src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs [hunk 1]
```diff
@@ -79,8 +79,7 @@ public async Task ExecuteAsync(FeedContext context)
             else
             {
                 context.Builder.AddProperty(context, null, "title", contentItem.DisplayText);
-
-                context.Builder.AddProperty(context, null, "description", bodyAspect.Body != null ? $"<![CDATA[{bodyAspect.Body?.ToString()}]]>" : String.Empty);
+                context.Builder.AddProperty(context, null, "description", new XCData(bodyAspect.Body?.ToString() ?? String.Empty).ToString());
```
Comment (scleaver) [inline src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs:82]: Same issue in the "else"  as for the CommonFeedItemBuilder here too.

**Review by scleaver — CHANGES_REQUESTED** [review]
There are still some issues inside the else when using context.Builder.AddProperty. See comments on the code.

File: src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs (hunk 1 above)
Comment (scleaver) [reply src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs:82]: Also no need to out in empty CDATA if the body is empty.

**Comment by hishamco** [comment]
@scleaver let me complete the unit tests and check your comment about `Builder.AddProperty()`

File: src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs:82]: BTW I think there's no issue with `<![CDATA[]]>` I see this quite often

Comment (scleaver) [reply src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs:82]: > BTW I think there's no issue with `<![CDATA[]]>` I see this quite often

Correct there is no issue... personal preference ;)

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lists/Feeds/ListFeedQuery.cs:82]: It is better to have `<![CDATA[]]>` than not, as it enforce defining which type of data is contained in that parameter even if it's empty. If you use a "" value in that attribute then it means that the returned content could vary of type.

**Comment by hishamco** [comment]
@scleaver you could try the latest changes

File: test/OrchardCore.Tests/Modules/OrchardCore.Contents/Feeds/RssFeedTests.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                DisplayText = "News",
+                PublishedUtc = DateTime.UtcNow
+            });
+
+            // Act
+            await commonFeedItemBuilder.PopulateAsync(feedContext);
+
+            // Assert
+            Assert.Contains($"<![CDATA[{bodyContent}]]>", feedContext.Response.Items[0].Element.ToString());
```
Comment (deanmarcussen) [inline test/OrchardCore.Tests/Modules/OrchardCore.Contents/Feeds/RssFeedTests.cs:50]: This looks generally good.

But the test does not test the issue. 
Where is the test that checks whether the elements have been html entity escaped or not?

**Review by Skrypt — CHANGES_REQUESTED** [review]
Assert that this : `<description><![CDATA[<p>My amazing description.</p>]]></description>`
Never becomes this : `<description>&lt;![CDATA[&lt;p&gt;My amazing description.&lt;/p&gt;]]&gt;</description>`

After that you can merge.

**Comment by hishamco** [comment]
For sure @Skrypt, I will do it now

File: test/OrchardCore.Tests/Modules/OrchardCore.Contents/Feeds/RssFeedTests.cs (hunk 2 above)
Comment (hishamco) [reply test/OrchardCore.Tests/Modules/OrchardCore.Contents/Feeds/RssFeedTests.cs:50]: @deanmarcussen I just checked whether the `<![CDATA[]>` becomes `&lt;![CDATA[&lt;]]]&gt;` or not ;) I will do the assert that requested by @Skrypt

**Pull request merged**